ros2topic/ros2topic/verb/pub.py

```diff
@@ -53,6 +54,7 @@ class PubVerb(VerbExtension):
     """Publish a message to a topic."""
 
     def add_arguments(self, parser, cli_name):
+        add_direct_node_arguments(parser)
```

Comment:
@ivanpauno nit: `help` string for `--spin-time` is somewhat misleading in this case:

https://github.com/ros2/ros2cli/blob/edbc1600aa161845c58138bc18907bd873670a54/ros2cli/ros2cli/node/direct.py#L91-L94

Reply:
why? the goal is exactly to wait for discovery to happen.
It might be less obvious that in the introspection commands, but it seems clear to me.

Maybe: `
'Spin time in seconds to wait for discovery, including endpoint matching if applies (only applies when '           'not using an already running daemon)'`?

Reply:
Well, actually you're right, in this case we would be spinning twice which is unexpected.
I will take a look again tomorrow, maybe I will go ahead with the approach commented above https://github.com/ros2/ros2cli/pull/641#issuecomment-848968007.

Reply:
Yeah, it seems like a "wait for N subscriptions" option would be more robust.

Reply:
> Well, actually you're right, in this case we would be spinning twice which is unexpected.

That, and the second spin not affected by the presence of the daemon.

Reply:
> it seems like a "wait for N subscriptions" option would be more robust.

i do think so too :+1: